pkg/dashboards/folders.go

```diff
@@ -797,6 +803,39 @@ func migrateToFolderStructure(myid int64) error {
 	return nil
 }
 
+func getFolderNestedCount(folderID string, myid int64) (*FolderNestedCount, error) {
+	structure, err := readCombinedFolderStructure(myid)
+	if err != nil {
+		return nil, fmt.Errorf("getFolderNestedCount: failed to read folder structure, err: %v", err)
+	}
+
+	if _, exists := structure.Items[folderID]; !exists {
+		return nil, fmt.Errorf("getFolderNestedCount: folder not found, folderID: %v", folderID)
+	}
+
+	count := &FolderNestedCount{Folders: 0, Dashboards: 0, Total: 0}
+	collectItemsCount(folderID, structure, count)
+
+	count.Total = count.Folders + count.Dashboards
+
+	return count, nil
+}
+
+func collectItemsCount(folderID string, structure *FolderStructure, count *FolderNestedCount) {
+	if childIDs, exists := structure.Order[folderID]; exists {
```

Comment:
a better coding pattern is to exit early instead of deep nested loops, makes code reading harder.
for e.g. I would do this:
```
childIDs, exists := structure.Order[folderID]
if !exists {
return 
}

for _, childID := range childIDs {
child, exists := structure.Items[childID]
   if  !exists {
         continue
  }
   if child.Type == ItemTypeFolder {
         count.Folders++
	 collectItemsCount(childID, structure, count)
   } else if child.Type == ItemTypeDashboard {
	 count.Dashboards++
		.....
```